src/inmanta/data/model.py

```diff
@@ -316,6 +319,31 @@ def level_from_enum_attribute(cls, v: object) -> object:
     kwargs: Dict[str, Optional[ArgumentTypes]] = {}
     timestamp: datetime.datetime
 
+    def dict(
```

Comment:
This is really ugly, and it could be avoided if the type of the `level` attribute was changed to `StrLogLevel`.  But this would break the api of `resource_deploy_done`.  Any suggestion welcome

Reply:
Could you elaborate on why this is required? I don't immediately what's changed that would require this method override.

Reply:
It might not be after all, discussion going on there: https://github.com/inmanta/inmanta-core/pull/3199#discussion_r699166198